plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -208,6 +212,8 @@ func (monitor *DirectoryMonitor) read(filePath string) {
 	if err != nil {
 		monitor.Log.Errorf("Error while reading file: '" + filePath + "'. " + err.Error())
 		monitor.filesDropped.Incr(1)
+		// Backward compatibility
+		selfstat.Register("directory_monitor", "files_dropped", nil).Incr(1)
```

Comment:
I don't think we should achieve backward compatibility like this. See my comments below.